### DISCUSSION TITLE
migrate from SQLite to MariaDB

### DISCUSSION THREAD

**Discussion by masterwishx** [FreshRSS/FreshRSS#4748] [Q&A] [answered]
can i migrate DB from SQlite to MariaDB after creation DB and some using FreshRSS ?

**Comment by Alkarex** [comment, marked as answer]
Yes, we have a command-line tool to export from any database to SQLite, and also to import from SQLite to any database.

See https://github.com/FreshRSS/FreshRSS/blob/edge/cli/README.md

It is not doing a full migration at the moment though, so you need to do a few things by hand.

Concretely, to change database, you need to:
1) Back-up your existing `./freshrss/data/` folder, which contains the SQLite files for each user
2) Reinstall FreshRSS from scratch, specifying your new database (I recommend PostgreSQL*)
3) Recreate the users (there is also a command-line for that if needed, otherwise from the Web UI)
4) For each users, use the `./cli/import-sqlite-for-user.php` command-line
    * `./cli/import-sqlite-for-user.php --user alice --force-overwrite --filename /backup/freshrss/data/users/alice/db.sqlite`
5) Optionally override `./freshrss/data/users/*/config.php` with the ones of your backup to transfer user preferences.

----

\* PostgreSQL is required for some features such as [text search indexing](https://freshrss.github.io/FreshRSS/en/admins/DatabaseConfig.html)
